cmd/tools/rest/swag.go

```diff
@@ -311,6 +318,65 @@ func readSwagger(args Args) (ontap, error) {
 	return ontap{apis: names, swagger: ontapSwag, collectionsApis: collectionsApis}, nil
 }
 
+// fixSwagger attempts to fix some known errors in ONTAP's swagger yaml file
+// that cause go-openapi/spec to fail. The errors are found in visitNode
+func fixSwagger(path string, b []byte) (spec.Swagger, error) {
+	var nodes map[string]interface{}
+	err := json.Unmarshal(b, &nodes)
+	if err != nil {
+		fmt.Printf("error unmarshalling %s into map err=%+v\n", args.SwaggerPath, err)
+		return spec.Swagger{}, err
+	}
+
+	// visit every node correcting known issues then try again
+	visitNode(nodes)
+	var ontapSwag spec.Swagger
+	nb, err := json.Marshal(nodes)
+	if err != nil {
+		fmt.Printf("error marshalling adjusted swagger file=[%s] to json err=%+v\n", args.SwaggerPath, err)
+		return spec.Swagger{}, err
+	}
+	err = json.Unmarshal(nb, &ontapSwag)
+	if err == nil {
+		// the fixes worked, write out the changes
+		out, err := os.Create(path)
+		if err != nil {
+			return spec.Swagger{}, fmt.Errorf("unable to create %s to save swagger.yaml", path)
+		}
+		defer silentClose(out)
+		_, err = io.Copy(out, bytes.NewReader(nb))
+		if err != nil {
+			return spec.Swagger{}, fmt.Errorf("error while saving mutated swagger to %s err=%w\n", path, err)
+		}
+	}
+	return ontapSwag, err
+}
+
+// visitNode corrects type errors in ONTAP's swagger.yaml file
+// the properties listed in the case statement below should have a type
+// of int instead of string - correct that so the yaml is valid and parses
+func visitNode(v interface{}) {
+	switch vv := v.(type) {
+	case map[string]interface{}:
+		for k, val := range vv {
+			switch k {
+			case "maxLength", "minLength", "maximum", "minimum", "minItems", "maxItems":
+				if s, ok := val.(string); ok {
+					num, err := strconv.Atoi(s)
+					if err == nil {
+						vv[k] = num
+					}
+				}
```

Comment:
may be we should use some array at line 363 and check with contains here as this list may grow in future.